libraries/node-core-library/src/Terminal/Terminal.ts

```diff
@@ -41,6 +44,19 @@ export class Terminal implements ITerminal {
     }
   }
 
+  /**
+   * {@inheritdoc ITerminal.setStatus}
+   */
+  public setStatus(...messageParts: (string | IColorableSequence)[]): void {
+    if (this._hasRenderedStatus) {
+      readline.clearLine(process.stdout, 0);
+      readline.cursorTo(process.stdout, 0);
```

Comment:
Putting this on the raw `Terminal` is a bit sketchy since terminal supports writing to any arbitrary destination, not just process.stdout.